util/omrutil/j9memclr.c

```diff
@@ -48,9 +48,19 @@ struct IHAPSA {
 static int isZ10orGreater = -1;
 #endif
 
+#if defined(J9ZOS390) || (defined(LINUX) && defined(S390))
+#include <stdlib.h>
+#endif /* defined(J9ZOS390) (z/OS) || (defined(LINUX) && defined(S390)) (zLinux) */
+
 void
 OMRZeroMemory(void *ptr, uintptr_t length)
 {
+#if defined(J9ZOS390) || (defined(LINUX) && defined(S390))
+static int pfd_to_zero_object_heap_memory = -1;
+if (NULL != getenv("PFD_TO_ZERO_OBJECT_HEAP_MEMORY_ENABLED"))
+	pfd_to_zero_object_heap_memory = 1;
+#endif /* defined(J9ZOS390) (z/OS) || (defined(LINUX) && defined(S390)) (zLinux) */
```

Comment:
- This will access the environment on every call to `OMRZeroMemory`. It would be better to call `getenv` in the declaration of the static, so it is called just once.
- By convention we don't use underscores in variable names, maybe something shorter like `usePFD` would be better (you can choose a name you like :))
- The environment variable should be prefixed with `OMR`, and again maybe something shorter would be a little more accessible.
- In the comment, no need to say how the ifdefs map to platforms, it's a little obvious.
- indentation is wrong, and if statements should always have {braces}.

Here's how I would write it:

```c++
#if defined(J9ZOS390) || (defined(LINUX) && defined(S390))
	static BOOLEAN usePFD = NULL != getenv("OMR_USE_PFD");
#endif /* defined(J9ZOS390) || (defined(LINUX) && defined(S390)) */
```


What does PFD mean?

Reply:
> This will access the environment on every call to OMRZeroMemory. It would be better to call getenv in the declaration of the static, so it is called just once.

Unfortunately, I can't do initialization at the same time as declaration as `c` doesn't consider `NULL != getenv("OMR_USE_PFD")` as a constant

> What does PFD mean?

PFD refers to using software to prefetch instructions. We have been using it in our assembly routine `j9memclrz10` https://github.com/eclipse/omr/blob/master/util/omrutil/unix/zos/64/j9memclrz10_64.s.